### DISCUSSION TITLE
ArgoCD and separation of environments

### DISCUSSION THREAD

**Discussion by brewerslack** [renovatebot/renovate#29683] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub, renovate:37.398.2

### Please tell us more about your question or problem

Hello! I'm trying to get my renovate bot to create a single PR per helm chart dependency per environment in my ArgoCD applications. I know that's a mouthful so an example might be easier. Say for example I have an instance of `ingress-nginx` running in both prod and dev. These are in separate directories running the same targetRevision in an ArgoCD application. Renovate in its current configuration will see that we have two instances of `ingress-nginx` and create a single PR to upgrade both of them. How can I get renovate to create 2 PRs, one for dev `ingress-nginx`, and one for prod? I've tried using `groupName` and `groupSlug`, but that just groups every dependency into one environment wide PR. I'd like this to be pretty granular 1 PR per 1 dependency per 1 environment. Thanks for the help!

chart spec:

```
spec:
  destination:
    namespace: ingress-basic
    server: {{ .Values.appDestinationServer }}
  source:
    repoURL: 'https://kubernetes.github.io/ingress-nginx'
    targetRevision: 4.10.0
    chart: ingress-nginx
```

Currently I have a configuration that looks like this:

```
{
      "repositories": ["reponamehere"],
      "argocd": {
        "fileMatch": ["^.*/app-.*\\.ya?ml$"]
      },
      "packageRules": [
        {
          "managers": ["argocd"],
          "paths": ["**/*dev*/**"],
        },
        {
          "managers": ["argocd"],
          "paths": ["**/*staging*/**", "**/*stg*/**"],
        },
        {
          "managers": ["argocd"],
          "paths": ["**/*prod*/**"],
        },
        {
          "managers": ["argocd"],
          "paths": ["**/*operations*/**"],
        },
        {
          "managers": ["kustomize", "github-actions"],
          "enabled": false
        }
      ],
      "timezone": "America/Los_Angeles",
      "rebaseWhen": "behind-base-branch",
      "automerge": false,
      "dependencyDashboard": true,
      "prHourlyLimit": 20
    }
```

**Comment by rarkins** [comment, marked as answer]
Try setting additionalBranchPrefix="{{parentDir}}"

Search other discussions for those terms for more details
